### DISCUSSION TITLE
IdMask image only shows upper half & Cloth not appearing in depth and amodal mask images

### DISCUSSION THREAD

**Discussion by hyhalexis** [empriselab/RCareWorld#111] [Help - Programming]
I am using the dressing env and want to check out different kinds of camera data. Here's my code for getting camera data:
   
>     camera_hand.GetRGB(512, 512)
>     env.step()
>     rgb = np.frombuffer(camera_hand.data["rgb"], dtype=np.uint8)
>     rgb = cv2.imdecode(rgb, cv2.IMREAD_COLOR)
>     cv2.imwrite("rgb.png", rgb)

>     camera_hand.GetID(512, 512)
>     env.step()
>     id_map = np.frombuffer(camera_hand.data["id_map"], dtype=np.uint8)
>     id_map = cv2.imdecode(id_map, cv2.IMREAD_COLOR)
>     cv2.imwrite("id_map.png", id_map)

>     camera_hand.GetAmodalMask(782563, 512, 512)
>     env.step()
>     amodal_mask = np.frombuffer(camera_hand.data["amodal_mask"], dtype=np.uint8)
>     amodal_mask = cv2.imdecode(amodal_mask, cv2.IMREAD_COLOR)
>     cv2.imwrite("amodal_mask.png", amodal_mask)

>     camera_hand.GetDepth(0.1, 2.0, 512, 512)
>     env.step()
>     depth = np.frombuffer(camera_hand.data["depth"], dtype=np.uint8)
>     depth = cv2.imdecode(depth, cv2.IMREAD_COLOR)
>     cv2.imwrite("depth.png", depth)

However, only the rgb image looks fine. The cloth is missing from the depth image, and it seems that the amodal mask does not recognize the cloth. The Id mask image also only shows the upper half.
![rgb](https://github.com/user-attachments/assets/1fd421c2-5465-4728-a2d8-35a2e6149ddc)
![id_map](https://github.com/user-attachments/assets/57b6bfce-cc52-4d64-87c0-fa62f33c33a7)
![amodal_mask](https://github.com/user-attachments/assets/76a859da-207c-4199-9a57-5fa69f4a9c0a)
![depth](https://github.com/user-attachments/assets/ccba024c-95fa-485e-ab8e-284bd5246fe9)

**Comment by shxingch** [comment]
HI @hyhalexis ! The issues with these camera APIs have been fixed! You can try it again.
